### DISCUSSION TITLE
Recommended way to run Nicotine+ in debug on Windows

### DISCUSSION THREAD

**Discussion by jackBonadies** [nicotine-plus/nicotine-plus#2292] [General]
Hello,

What is the recommended way to run Nicotine+ in debug on Windows?  For Linux its as simple as installing any dependencies with pip and running "python nicotine".  For Windows, one cannot install pygobject through pip.  The only way I can find to install pygobject is through msys2, but I then got a host of other errors (though not sure if it was something I was doing incorrectly on my end).  It would also be nice to add the recommended way to the documentation (assuming that the task would in fact need documentation and its not just due to mistakes on my end).

Thanks!

**Comment by mathiascode** [comment]
- If you need development builds, but don't need to modify the source code: https://nicotine-plus.org/doc/TESTING.html#windows
- If you need to modify the source code: https://nicotine-plus.org/doc/PACKAGING.html#windows
- If you need to enable debug logging (release and development builds):
  1. Show the log pane (Menu -> View -> Show log history pane)
  2. Right-click the log pane -> Log categories -> Enable the categories you want (keep in mind that this will affect performance, especially if you search for popular files)
  3. If you want to log to file, Preferences -> Logging -> Log debug messages to file

**Reply by jackBonadies** [reply]
Maybe this is my ignorance in python, but I want to run the .py file in the interpreter so that I can set breakpoints (either with IDE or pdb.set_trace()).  I can make an executable with my changes with cx_Freeze, but cannot figure out a way to debug the resulting executable - if I build the .exe with pdb.set_trace lines included, it will successfully break there but then pdb fails to find the associated source code.

**Reply by mathiascode** [reply]
You don't have to build the executable if you don't need it. You should be able to run `python nicotine`, like you would on Linux.
